### ISSUE TITLE
[Issue]: Inpaint sketch resizing image on canvas without being prompted

### ISSUE THREAD

**Issue by mark-wd** [vladmandic/automatic#870]
### Issue Description

Following from #860

Updated to the latest build and the problem persists.

Digging deeper, I've found that this behavior happens during the following workflow:

- Generate a 512wX768h image in txt2img, my batch count was 4.
- Choose one of them and click the "->IMAGE" button below to send to img2img
- On the img2img tab, click the "Inpaint sketch" button below the canvas
- At this point, computer resource usage spikes way up. The resulting image on the canvas is actually resized to be way bigger
- Browser slows down massively and any further work is halted due to the size of the image. If it manages to be processed, it results in the error.

Behavior seen on Win11/Firefox

**Comment by vladmandic** [comment]
i still cannot reproduce. 
can you open browser inspector window (F12) and select image element - what does it show for its actual canvas elements?
example - you can see in HTML that all canvas elements have correct width=512 height=768

![image](https://github.com/vladmandic/automatic/assets/57876960/90f51e32-9b85-4234-90d1-afcbe5e1a3fa)

**Comment by crazygonuts** [comment]
This is also occuring for me - copying an image from the sketch tab to the img2img tab upscales the image by 1.5x

the size of the image in the html image container under the sketch tab is correct

**Comment by mark-wd** [comment]
Now that you mention it, it's actually all screwed:

![Screenshot 2023-05-12 013421](https://github.com/vladmandic/automatic/assets/31883671/76d0b9f8-baca-4c41-b8e0-1cecf2c41ac7)

I am doing this on a monitor with a pretty huge resolution... maybe it's capturing that as a basis or something of the sort?

**Comment by vladmandic** [comment]
@mark-wd yeah, that's clearly messed up, no question is the issue real or not. i wish i could reproduce, no idea why i cant.
two questions:
- can you reproduce on chrome as well?
- are you using windows resolution scaling?

**Comment by mark-wd** [comment]
New tests:
- In Chrome latest: same result
- 1080p monitor instead of 4K: same result

One important thing is that Inpaint itself works fine. Maybe this has to do with the specific way in which Inpaint Sketch is invoked?

Well, anyway, this is not super high priority... still good if you can add it to your backlog.

**Comment by vladmandic** [comment]
Yup, I'll keep it in the backlog...

**Comment by vladmandic** [comment]
can you check if the issue still exists?

**Comment by vladmandic** [comment]
any updates?

**Comment by raisindetre** [comment]
Just chiming in that I'm getting the same issue/effect in A1111 v1.6.0 on OSX ARM M2. Slightly different workflow (sending to img2img from PNG Info, disabling all extension settings, clicking on Inpaint Sketch. Massive slow down of browser, canvas minimally responsive and huge canvas image sizes reported in HTML.
<img width="589" alt="inpaint-sketch-canvas" src="https://github.com/vladmandic/automatic/assets/10620977/61a91796-4374-42b2-85a7-c33e2cff3e1a">

I suspect this is related to screen resolution in some way. I'm running on a Macbook Pro with a Retina dislplay, so 3024 x 1964 at 254 pixels per inch with I think some sort of display scaling.

**Comment by raisindetre** [comment]
> Just chiming in that I'm getting the same issue/effect....

This was in latest Firefox and Chrome BTW. I tried changing display resolution to 960x600 and restarting A1111 but this time the image disappeared in the Inpainting Sketch canvas and got an unhappy face icon.

**Comment by raisindetre** [comment]
I have retested, this time setting Firefox application zoom scale to 50% and Inpaint Sketch works without fault so this seems to be the workaround at present until the code can address high-res displays as the img2img Inpaint canvas presently seems to do alright.

**Comment by vladmandic** [comment]
Thanks for the update...

**Comment by AlexeyWP** [comment]
Had about the same issue. It was caused display settings: here wa set 125%. Returned it back to 100%. Now it works OK.
![gh1](https://github.com/vladmandic/automatic/assets/79204812/29efbe66-70ef-4eb6-9bc1-e648daf9fffa)

**Comment by alastaira** [comment]
Can confirm I just experienced the same issue.

- Clicking the "Send to Inpaint Sketch" button to send a 512x512px image from img2img makes browser (Firefox 118.0.1) very unresponsive, and subsequent attempts to use inpaint sketch crash out with a DecompressionBombError.
- On inspecting the HTML, the canvas elements have been created at 718x8092px  (the mask is created at 718x4096px)
- Send to inpaint (not inpaint sketch) works as expected
- Windows Display had been set to 200%. Changing this to 100% appears to fix the problem (although makes my PC otherwise unusable!) 

![image](https://github.com/vladmandic/automatic/assets/901056/6d26ba44-6a9f-43ab-9e95-364d704b0417)
![image](https://github.com/vladmandic/automatic/assets/901056/f7a0e65f-afa2-48f4-8af7-1b2dc49a36cf)
![image](https://github.com/vladmandic/automatic/assets/901056/b8980930-fe3a-4ddb-9f2c-f786c80f4e12)

**Comment by vladmandic** [comment]
linking to actual upstream issue https://github.com/gradio-app/gradio/issues/5858

**Comment by vladmandic** [comment]
somewhat disappointing response from gradio team: https://github.com/gradio-app/gradio/issues/5858#issuecomment-1755422024
if you want to add to issue, please comment to gradio directly.

**Comment by vladmandic** [comment]
work on new canvas has started and all related issues will be tracked in #4358
